From 2591c26efe26711de64bcb3a0997cc1846cb3ea2 Mon Sep 17 00:00:00 2001
From: Mika Laitio <lamikr@gmail.com>
Date: Sat, 22 Mar 2025 08:51:34 -0700
Subject: [PATCH 5/5] rocm clang version check fix

clang in rocm sdk 6.3.3 returns 18.0git as a version.
fix this by adding a check which converts it to
18.0 without throwing an exception

Signed-off-by: Mika Laitio <lamikr@gmail.com>
---
 torch/utils/cpp_extension.py | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/torch/utils/cpp_extension.py b/torch/utils/cpp_extension.py
index 9317ada539f..3311d3a5b54 100644
--- a/torch/utils/cpp_extension.py
+++ b/torch/utils/cpp_extension.py
@@ -454,6 +454,15 @@ def get_compiler_abi_compatibility_and_version(compiler) -> tuple[bool, TorchVer
         warnings.warn(f'Error checking compiler version for {compiler}: {error}')
         return (False, TorchVersion('0.0.0'))
 
+    # manage version 18.0git returned by clang in rocm sdk 6.3.3
+    for ii in range(len(version)):
+        try:
+            version_item = version[ii]
+            test = int(version_item)
+        except ValueError:
+            version_item = '0'
+            version[ii] = version_item
+        print("version: " + version_item)
     if tuple(map(int, version)) >= minimum_required_version:
         return (True, TorchVersion('.'.join(version)))
 
-- 
2.48.1

